### ISSUE TITLE
Slug change no warning

### ISSUE THREAD

**Issue by jamesgillen** [torchbox/wagtail#528]
When you change the name of a page, it would be nice if you get a warning message telling the user  that the slug will change, or even better, get a choice if you want it to change or not.

**Comment by davecranwell** [comment]
This report is mostly correct. 

The current logic is that if the slug of the page is identical to the title field, then changing the title also changes the slug. The (false) assumption being that by having the two fields the same, you're intending the two fields to always be tied.

If however the slug is different to the title, say if the user changed the slug to be more seo-friendly, then the slug will not change when the title is edited. This behaviour only remains for as long as the slug and title differ: if the title is changed to match the slug, then the page is saved, then the title is edited again, the slug will also now change.

A better behaviour would probably be for the current behaviour to remain only for as long as the page remains un-saved i.e only for brand new, unsaved pages. Once saving has occurred first time, even if the page remains unpublished, the behaviour is disabled on that page forever. The javascript which does the matching was really only added to make creating pages quick, avoiding the validation error of trying to submit a page where you'd forgotten to fill in fields on the "Promote" tab.

**Comment by chrxr** [comment]
Thanks @jsma. Agreed that this is important to fix. Assigning to @kaedroho.

**Comment by tomdyson** [comment]
@bertrandbordage, could you look at this, please?

**Comment by BertrandBordage** [comment]
@tomdyson Sure.
If I understand well, the idea would be the same as setting `ModelAdmin.get_prepopulated_fields` (if we were using django admin) to something like:
```python
def get_prepopulated_fields(request, obj=None):
    if obj is None:
        return {'slug': ('title',)}
    return {}
```

Am I right?

**Comment by tomdyson** [comment]
@BertrandBordage yes, but in JavaScript.

**Comment by BertrandBordage** [comment]
@tomdyson Wouldn’t it be a good occasion to add something like `prepopulated_fields` to wagtail’s admin?

**Comment by tomdyson** [comment]
@BertrandBordage, I defer to @gasman and @kaedroho.

**Comment by lee-kagiso** [comment]
We are also running into this problem.

Content editors publish a page, tweet and fb the links out, edit the title (and break the link as such), and then we serve up 404s to everyone.

**Comment by timheap** [comment]
This has caught us out as well. Changing the slug automatically - certainly when it is hidden away on a separate tab - confuses our users and even us developers.

**Comment by tomdyson** [comment]
@BertrandBordage did most of the work to fix this in #1148 but it needs the changes @gasman suggested here https://github.com/torchbox/wagtail/pull/1148#issuecomment-97772797. Can anyone take this on, ideally in the next few days before the 1.2 feature freeze?
